src/oc/lib/api/common.clj

```diff
@@ -187,17 +187,49 @@
    If no Authorization headers are supplied return nil."
   [headers passphrase]
   (when-let [token (get-token headers)]
-    (if (jwt/valid? token passphrase)
-      {:jwtoken token
-       :user (:claims (jwt/decode token))}
-      {:jwtoken false})))
+    (cond
+
+     (jwt/valid? token passphrase)
+     {:jwtoken token
+      :user (:claims (jwt/decode token))}
+
+     ;; identity token
+     (and (jwt/check-token token passphrase)
+          (:id-token (:claims (jwt/decode token))))
+     {:jwtoken false :id-token token}
+
+     :default
+     {:jwtoken false})))
+
+(defn allow-id-token
+  "Allow options request. Allow jwtoken. Allow id token. Allow anonymous."
+  [ctx]
+  (cond
+
+   (= (-> ctx :request :request-method) :options)
+   true
+
+   (:jwtoken ctx)
+   true
+
+   (:id-token ctx)
+   true
+
+   (nil? (:jwtoken ctx))
+   true
```

Comment:
Not sure what this means, if you are here you don't have an id-token, so this is like allowing anonymous, if you don't have an id-token and don't have a jwt you can see this? You shouldn't

Reply:
This is allowing anonymous usage.